metricbeat/module/mysql/query/query.go

```diff
@@ -0,0 +1,132 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/*
+Package status fetches MySQL server status metrics.
+
+For more information on the query it uses, see:
+http://dev.mysql.com/doc/refman/5.7/en/show-status.html
+*/
+package query
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/metricbeat/helper/sql"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+func init() {
+	mb.Registry.MustAddMetricSet("mysql", "query", New,
+		mb.DefaultMetricSet(),
+	)
+}
+
+type query struct {
+	QueryNamespace string `config:"query_namespace"`
```

Comment:
Yeah, it's to avoid confusion with the metricset namespace. I got confused a couple of times myself while testing the module and I thought it was a bit too implicit to see. What I can do is to change the variable name inside the code but I think that the user facing yaml should remain as `query_namespace`.

Actually, I wanted to call the other `metricset_namespace` too :smile: The issue is that `namespace` is so widely used everywhere that it has way too many meanings.

Reply:
This is how it looks now: https://github.com/elastic/beats/pull/18955/files#diff-4315d136c22767eae25860e395f2798cR7